Kernel/Storage/AHCIPortHandler.cpp

```diff
@@ -14,6 +14,43 @@ NonnullRefPtr<AHCIPortHandler> AHCIPortHandler::create(AHCIController& controlle
     return adopt_ref(*new AHCIPortHandler(controller, irq, taken_ports));
 }
 
+NonnullRefPtr<AHCIPortHandler> AHCIPortHandler::create(AHCIController& controller, u8 irq, AHCI::MaskedBitField taken_ports, NonnullRefPtr<AHCIPort> (*ahci_port_creator)(const AHCIPortHandler&, volatile AHCI::PortRegisters&, u32 port_index))
+{
+    return adopt_ref(*new AHCIPortHandler(controller, irq, taken_ports, ahci_port_creator));
+}
+
+AHCIPortHandler::AHCIPortHandler(AHCIController& controller, u8 irq, AHCI::MaskedBitField taken_ports, NonnullRefPtr<AHCIPort> (*ahci_port_creator)(const AHCIPortHandler&, volatile AHCI::PortRegisters&, u32 port_index))
```

Comment:
Hmm, I really don't like how this thing works... we essentially pass a pointer (and not a `Function` like we should...) of a function. I'm almost sure we can do it in a better way, and not to pass a callback to some sort of creator.